site/docs/plugins/commands.md

```diff
@@ -5,10 +5,552 @@ next: false
+  "logout",
+  "End your session with the bot",
+  async (ctx) => {
+    await ctx.setMyCommands(loggedOutCommands);
+    await ctx.reply("Goodbye :)");
+  },
+);
+
+bot.use(loggedInCommands);
+bot.use(loggedOutCommands);
+
+// By default, users are not logged in,
+// so you can set the logged out commands for everyone
+await loggedOutCommands.setCommands(bot);
+```
+
+```js [JavaScript]
+// Register the context shortcut
+bot.use(commands());
+
+const loggedOutCommands = new Commands();
+const loggedInCommands = new Commands();
+
+loggedOutCommands.command(
+  "login",
+  "Start your session with the bot",
+  async (ctx) => {
+    await ctx.setMyCommands(loggedInCommands);
+    await ctx.reply("Welcome! Session started!");
+  },
+);
+
+loggedInCommands.command(
+  "logout",
+  "End your session with the bot",
+  async (ctx) => {
+    await ctx.setMyCommands(loggedOutCommands);
+    await ctx.reply("Goodbye :)");
+  },
+);
+
+bot.use(loggedInCommands);
+bot.use(loggedOutCommands);
+
+// By default, users are not logged in,
+// so you can set the logged out commands for everyone
+await loggedOutCommands.setCommands(bot);
+```
+
+:::
+
+This way when a user calls `/login`, they'll have their commands list changed to contain only the `logout` command. Neat, right?
+
+If you want to prevent, for example, the commands contained in `loggedInCommands` from being callable after the user called `/logout`, you must implement it in your handlers with your own business logic.
+
+::: danger
+As stated in the [Telegram API documentation](https://core.telegram.org/bots/api#botcommand), command names can only be form out of:
+
+> 1-32 characters. Can contain only lowercase English letters, digits and underscores.
+
+Therefore calling `setCommands` or `setMyCommands` on upperCased commands will throw an exception. They can still be registered and used, but will never be displayed on the user menu as such.
+
+**Setting UpperCased command names is heavily discourage**
+:::
+
+**Be aware** that `SetCommands` and `SetMyCommands` only affects the commands displayed in the user's commands menu, and not the actual access to them. You will learn how to implement restricted command access in the [Scoped Commands](#scoped-commands) section.
+
+### Grouping Commands
+
+Since we can split and group our commands into different instances, it allows for a much more idiomatic command file organization.
+
+Let say we want to have developer-only commands. We can achieve that with the the following code structure:
+
+```ascii
+src/
+├─ commands/
+│  ├─ users.ts
+│  ├─ admin.ts
+├─ bot.ts
+├─ types.d.ts
+tsconfig.json
+```
+
+::: tip
+We are assuming your `tsconfig` file is well-set to resolve the types from `types.d.ts` and have resolved every necessary import.
+:::
+
+The following code group exemplify how we could implement a developer only command group, and update our Telegram client Command menu. Make sure you inspect the `admin.ts` and `user.ts` file-tabs.
```

Comment:
should we rename the `Commands` class to `CommandGroup`? I feel like it expresses what it is way better